src/SourceCode.Clay/Sha1.cs

```diff
@@ -108,20 +124,26 @@ public Sha1(in ReadOnlySpan<byte> span)
         public static Sha1 Hash(in string value)
         {
             if (value == null) return Zero;
-            // Note that length=0 should not short-circuit
+            if (value.Length == 0) return _empty;
 
             // Rent buffer
             var maxLen = Encoding.UTF8.GetMaxByteCount(value.Length); // Utf8 is 1-4 bpc
```

Comment:
Instead of renting a giant buffer here, it might be better to rent one `Environment.PageSize` and stream into Sha1.

Reply:
Done